sim/rogue/poisoned_knife.go

```diff
@@ -0,0 +1,71 @@
+package rogue
+
+import (
+	"time"
+
+	"github.com/wowsims/sod/sim/core"
+	"github.com/wowsims/sod/sim/core/proto"
+)
+
+func (rogue *Rogue) registerPoisonedKnife() {
+	if !rogue.HasRune(proto.RogueRune_RunePoisonedKnife) {
+		return
+	}
+
+	rogue.PoisonedKnife = rogue.RegisterSpell(core.SpellConfig{
+		ActionID:    core.ActionID{SpellID: int32(proto.RogueRune_RunePoisonedKnife)},
+		SpellSchool: core.SpellSchoolPhysical,
+		ProcMask:    core.ProcMaskMeleeOHSpecial,
+		Flags:       core.SpellFlagMeleeMetrics | core.SpellFlagIncludeTargetBonusDamage | SpellFlagBuilder | core.SpellFlagAPL,
+
+		EnergyCost: core.EnergyCostOptions{
+			Cost:   []float64{25, 22, 20}[rogue.Talents.ImprovedSinisterStrike],
+			Refund: 0.8,
+		},
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				GCD: time.Second,
+			},
+			CD: core.Cooldown{
+				Timer:    rogue.NewTimer(),
+				Duration: time.Second * 6,
+			},
+			IgnoreHaste: true,
+		},
+		ExtraCastCondition: func(sim *core.Simulation, target *core.Unit) bool {
+			return rogue.HasOHWeapon() && rogue.DistanceFromTarget >= 8
+		},
+		CastType: proto.CastType_CastTypeRanged,
+
+		DamageMultiplier: 1 +
+			0.02*float64(rogue.Talents.Aggression)*rogue.dwsMultiplier(),
+		CritMultiplier:   rogue.MeleeCritMultiplier(true),
+		ThreatMultiplier: 1,
+		// Cannot Miss
+		BonusHitRating: 100 * core.MeleeHitRatingPerHitChance,
+
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			rogue.BreakStealth(sim)
+			baseDamage := spell.Unit.OHNormalizedWeaponDamage(sim, spell.MeleeAttackPower()) + spell.BonusWeaponDamage()
+
+			// Cannot Miss, Dodge, or Parry as per spell flags
+			result := spell.CalcAndDealDamage(sim, target, baseDamage, spell.OutcomeMeleeSpecialNoBlockDodgeParry)
```

Comment:
Nevermind, it's a ranged ability with DefenseType Melee... talk about weird.

https://wago.tools/db2/SpellMisc?build=1.15.1.53181&filter[SpellID]=425012&page=1 - it has "Always hit", "No attack dodge", "No attack parry". Did you ever get _Blocks_ in testing?

It also has pretty much every "suppress proc"-Flag set available, so maybe it doesn't proc at all...

Reply:
I should test again, but I don't think I saw any Block.

Reply:
I can test for weapon procs later to confirm or deny the behavior